src/components/vm/vmMigrateDialog.jsx

```diff
@@ -0,0 +1,182 @@
+ */
+
+import cockpit from 'cockpit';
+import React, { useState } from 'react';
+import {
+    Alert,
+    Button,
+    Checkbox,
+    Flex,
+    Form,
+    FormGroup,
+    Modal,
+    Radio,
+    TextInput,
+    Tooltip
+} from '@patternfly/react-core';
+import { OutlinedQuestionCircleIcon } from "@patternfly/react-icons";
+
+import { migrateToUri } from '../../libvirt-dbus.js';
+import { isEmpty, isObjectEmpty } from '../../helpers.js';
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+
+const _ = cockpit.gettext;
+
+const DestUriRow = ({ validationFailed, destUri, setDestUri }) => {
+    return (
+        <FormGroup label={_("Destination URI")} fieldId="dest-uri-input"
+                   helperTextInvalid={validationFailed.destUri}
+                   validated={validationFailed.destUri ? "error" : "default"}>
+            <TextInput id='dest-uri-input'
+                       validated={validationFailed.destUri ? "error" : "default"}
+                       value={destUri}
+                       placeholder={cockpit.format(_("Example, $0"), "qemu+ssh://192.0.2.16/system")}
+                       onChange={setDestUri} />
+        </FormGroup>
+    );
+};
+
+const OptionsRow = ({ temporary, setTemporary }) => {
+    return (
+        <FormGroup label={_("Options")}
+                   hasNoPaddingTop>
+            <Checkbox id="temporary"
+                      isChecked={temporary}
+                      label={
+                          <Flex spaceItems={{ default: 'spaceItemsSm' }} alignItems={{ default: 'alignItemsCenter' }}>
+                              <span>{_("Move temporarily")}</span>
+                              <Tooltip position="top"
+                                       content={
+                                           <>
+                                               <p>{_("By default, the migrated VM config is removed from the source host, and saved persistently on the destination host. The destination host is considered the new home of the VM.")}</p>
+                                               <p>{_("If 'temporary' is selected, the migration is considered only a temporary move: the source host maintains a copy of the VM config, and the running copy moved to the destination is only transient, and will disappear when it is shutdown.")}</p>
+                                           </>}>
+                                  <OutlinedQuestionCircleIcon />
+                              </Tooltip>
+                          </Flex>
+                      }
+                      onChange={setTemporary} />
+        </FormGroup>
+    );
+};
+
+const StorageRow = ({ storage, setStorage }) => {
+    return (
+        <FormGroup label={_("Storage")}
+                   hasNoPaddingTop>
+            <Radio id="shared"
+                   name="storage"
+                   label={_("VM's storage is already shared with the  destination")}
```

Comment:
The strings are probably going to change again anyway, but this one has two spaces. These are annoying/confusing for translators, even though they are not actually rendered.